Content.Server/Medical/HealingSystem.cs

```diff
@@ -151,7 +151,10 @@ private bool TryHeal(EntityUid uid, EntityUid user, EntityUid target, HealingCom
         if (!TryComp<StackComponent>(uid, out var stack) || stack.Count < 1)
             return false;
 
-        if (!HasDamage(targetDamage, component))
+        if (!TryComp<BloodstreamComponent>(target, out var bloodstream))
+            return false;
+
+        if (!HasDamage(targetDamage, component) && !(bloodstream.BloodSolution.Volume < bloodstream.BloodSolution.MaxVolume && component.ModifyBloodLevel > 0))
```

Comment:
I'm a tad bad with reading code, but should `component.ModifyBloodLevel` be `healing.ModifyBloodLevel`?

Blood is not added to the bloodstream if  `healing.ModifyBloodLevel` is 0, if you look at line 73.

I'm unsure what `component.ModifyBloodLevel` references in this file...

Reply:
Now that I am looking at it, I am not sure either. I think it is the `HealingComponent` for `uid` but I might be wrong there.

Reply:
so essentially, component here is HealingComponent for the targetted entity (so this is fine)

and if i'm reading this correctly, if there's no damage and the bloodstream can't be modified (either because it's hit max volume or because the entity can't actually modify blood) then it'll do an early return with the given can't use message instead of performing the DoAfter

so i think this is fine